sql/parser/builtins.go

```diff
@@ -761,3 +773,73 @@ func datumToRawString(datum Datum) (string, error) {
 
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
+
+// Round rounds x to n digits.
+// Directly taken from CPython's `double_round` in
+// cpython/Object/floatobject.c.
```

Comment:
Actually, python has two versions: one on line 863 which uses `snprintf` and `strtod`, and one on line 931 which uses this math. Curiously, the `snprintf` version seems to be preferred, and the mathy version is only used as a "fallback" (it's unclear when that `#ifdef` is used). An example in http://bugs.python.org/issue1869 is `round(562949953421312.5, 1)`, which should return itself but turns into 562949953421312.62 in the mathy version. Maybe `strconv.FormatFloat()` is the way to go after all.